internal/provider/exemption_resource_test.go

```diff
@@ -0,0 +1,57 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccExemptionResource(t *testing.T) {
+	CheckAccTestsEnabled(t)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create + Read.
+			{
+				Config: providerConfig + `
+resource "adaptive-metrics_exemption" "test" {
+	metric = "test_tf_metric"
+	keep_labels = ["namespace"]
+}
+`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("adaptive-metrics_exemption.test", "metric", "test_tf_metric"),
+					resource.TestCheckResourceAttr("adaptive-metrics_exemption.test", "keep_labels.#", "1"),
+					resource.TestCheckResourceAttr("adaptive-metrics_exemption.test", "keep_labels.0", "namespace"),
+				),
+			},
+			// ImportState.
+			{
+				ResourceName:      "adaptive-metrics_exemption.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// The last_updated attribute does not exist in the
+				// aggregations API, therefore there is no value for it during
+				// import.
+				ImportStateVerifyIgnore: []string{"last_updated"},
+			},
+			// Update + Read.
+			{
+				Config: providerConfig + `
+resource "adaptive-metrics_exemption" "test" {
```

Comment:
Quite weird to have a dash in a resource name. It can cause syntax errors sometimes. I'd use an underscore

Reply:
Personally, I would even set the provider prefix to `grafana_adaptive_metrics_`

Reply:
Yep, I agree. When I changed `resp.TypeName` in the `Metadata` method of the provider though, everything else broke so I decided to leave it for now and figure it out later.

Reply:
👍 It's ok for now then. Let's figure it out before release though